ml-agents-envs/mlagents/envs/brain.py

```diff
@@ -5,19 +5,33 @@
 from mlagents.envs.communicator_objects.agent_info_pb2 import AgentInfoProto
 from mlagents.envs.communicator_objects.brain_parameters_pb2 import BrainParametersProto
 from mlagents.envs.timers import hierarchical_timer, timed
-from typing import Dict, List, Optional
+from typing import Dict, List, NamedTuple, Optional
 from PIL import Image
 
 logger = logging.getLogger("mlagents.envs")
 
 
+class CameraResolution(NamedTuple):
+    height: int
+    width: int
+    gray_scale: bool
+
+    @property
+    def num_channels(self) -> int:
+        return 1 if self.gray_scale else 3
```

Comment:
I almost want to say CameraResolution shouldn't be aware of whether it's grey scale or not, just how many channels it has. Seems more generalizable that way. We could pass it in 1 or 3 on creation from the proto instead - what do you think?

Reply:
Agreed, but that would be a much bigger change (would require communicator proto changes), and will (hopefully) come for free with the sensor refactor.

Reply:
OK, if it's coming anyways in the sensor refactor, no sense in doing it twice.

Reply:
Yeah, should be coming then, and also gives me some more confidence while doing the refactoring.